providers/dynamodb/shedlock-provider-dynamodb/src/main/java/net/javacrumbs/shedlock/provider/dynamodb/DynamodbLockProvider.java

```diff
@@ -0,0 +1,306 @@
+                                    .withWriteCapacityUnits(writeCapacityUnits(minLockTime)))
+                    .withTableName(tableName);
+
+            dynamoDB.createTable(createTableRequest).waitForActive();
+
+        }
+
+        private Long writeCapacityUnits(Duration minLockTime) {
+            Double queryPerSecond = Math.ceil(1 / (minLockTime.toMinutes() * 60)) + 1;
+            return queryPerSecond.longValue();
+        }
+
+        private Long readCapacityUnits(Duration minLockTime, long concurrentProcessCount) {
+            float queryDuration = minLockTime.toMinutes() * 60 / concurrentProcessCount;
+            Float queryPerSecond = (1 / queryDuration) + 1;
+            return queryPerSecond.longValue();
+        }
+
+        private Date now() {
+            return new Date();
+        }
+
+        @Override
+        public boolean updateRecord(LockConfiguration lockConfiguration) {
+
+            DynamoDB dynamoDB = new DynamoDB(amazonDynamoDB);
+            Table table = dynamoDB.getTable(tableName);
+            UpdateItemSpec spec = new UpdateItemSpec()
+                    .withPrimaryKey(NAME, lockConfiguration.getName())
+                    .withUpdateExpression(
+                            String.format("set %s=:%s, %s=:%s, %s=:%s",
+                                    LOCKED_BY, LOCKED_BY, LOCK_UNTIL, LOCK_UNTIL, LOCKED_AT, LOCKED_AT))
+                    .withConditionExpression(
+                            String.format("%s <= :%s", LOCK_UNTIL, LOCKED_AT))
+                    .withValueMap(
+                            new ValueMap()
+                                    .withString(":" + LOCKED_BY, lockConfiguration.getName())
+                                    .withLong(":" + LOCK_UNTIL, lockConfiguration.getLockAtMostUntil().toEpochMilli())
+                                    .withLong(":" + LOCKED_AT, Instant.now().toEpochMilli()))
+                    .withReturnValues(ReturnValue.ALL_OLD);
+
+            try {
+                UpdateItemOutcome updateItemOutcome = table.updateItem(spec);
+                return (updateItemOutcome.getItem() != null);
+            } catch (ConditionalCheckFailedException cfailed) {
+                return false;
+            }
+
+
+        }
+
+        @Override
+        public void unlock(LockConfiguration lockConfiguration) {
+            // Set lockUtil to now or lockAtLeastUntil whichever is later
+
+            DynamoDB dynamoDB = new DynamoDB(amazonDynamoDB);
+            Table table = dynamoDB.getTable(tableName);
+
+            UpdateItemSpec spec = new UpdateItemSpec()
+                    .withPrimaryKey(NAME, lockConfiguration.getName())
+                    .withUpdateExpression(String.format("set %s=:%s", LOCK_UNTIL, LOCK_UNTIL))
+                    .withValueMap(new ValueMap()
+                            .withLong(":" + LOCK_UNTIL,
+                                    Instant.now().isAfter(lockConfiguration.getUnlockTime())
```

Comment:
Why it is not possible to set lockConfiguration.getUnlockTime() in both cases?

Reply:
You are right, setting current time is better, I will fix it in other providers as well